src/core/query/woql_compile.pl

```diff
@@ -1072,12 +1072,14 @@
     % This probably makes all super user checks redundant.
     !.
 assert_pre_flight_access(Context, AST) :-
-    find_resources(AST,
-                   (Context.default_collection),
-                   (Context.filter),
-                   (Context.write_graph),
-                   Read,
-                   Write),
+    do_or_die(
+        find_resources(AST,
+                       (Context.default_collection),
+                       (Context.filter),
+                       (Context.write_graph),
+                       Read,
+                       Write),
+        error(find_resource_pre_flight_failure_for(AST))),
```

Comment:
Is this an error that we should catch and report? If so, we should update `api_error.pl` and give a meaningful error message.

Reply:
Added